pkg/controller/kieapp/defaults/defaults_test.go

```diff
@@ -7491,3 +7508,342 @@ func TestRhpamKieserverEnvWithoutDecisionsOnlyEnabled(t *testing.T) {
 	result, _ := strconv.ParseBool(getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_DECISIONS_ONLY"))
 	assert.False(t, result)
 }
+
+func TestKieServerWithEdgeTerminationEnabledHttp(t *testing.T) {
+	privateKey := "|-\n      -----BEGIN PRIVATE KEY-----\n      [...]\n      -----END PRIVATE KEY-----"
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+		},
+		Spec: api.KieAppSpec{
+			Environment: api.RhpamProductionImmutable,
+			Objects: api.KieAppObjects{
+				Servers: []api.KieServerSet{
+					{
+						TerminationRoute: &api.TerminationRoute{
+							Protocol:      constants.HttpProtocol,
+							Key:           privateKey,
+							Certificate:   privateKey,
+							CaCertificate: privateKey,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	env := getAndCheckEnv(t, cr)
+	checkTLSAndRoute(t, env.Servers[0].Routes[0].Spec.TLS, constants.EdgeTermination, env.Servers[0].Routes[0])
+	assert.Equal(t, intstr.IntOrString{Type: 1, IntVal: 0, StrVal: constants.HttpProtocol}, env.Servers[0].Routes[0].Spec.Port.TargetPort)
+}
+
+func TestKieServerWithEdgeTerminationEnabledHttps(t *testing.T) {
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+		},
+		Spec: api.KieAppSpec{
+			Environment: api.RhpamProductionImmutable,
+			Objects: api.KieAppObjects{
+				Servers: []api.KieServerSet{
+					{
+						TerminationRoute: &api.TerminationRoute{
+							Protocol: constants.HttpsProtocol,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	env := getAndCheckEnv(t, cr)
+	checkTLSAndRoute(t, env.Servers[0].Routes[0].Spec.TLS, constants.EdgeTermination, env.Servers[0].Routes[0])
```

Comment:
this checks is confusing, shouldn't edge termination be http? why are you setting protocol as https and checking for edge termination?
shouldn't be passtrough?

Reply:
IS to test if askingedge (enableEdge) and setting the worn protocol via CR the code set properly edge termination and http as a protocol